ansible/roles/ci/code_style_check/common_resources/files/files_by_extension.py

```diff
@@ -15,13 +15,15 @@
 # with this program. If not, see <http://www.gnu.org/licenses/>.
 
 import sys
+import re
 import os
 import argparse
 
 
+LINT_IGNORE_FILE = "lint_exclusions.cfg"
 CPP_HEADERS = [".cpp", ".c", ".h", ".hpp"]
 PYTHON_HEADERS = ["#!/usr/bin/env python", "#! /usr/bin/env python", "#!/usr/bin/python", "#! /usr/bin/python",
-                       "#!/usr/bin/env python3", "#! /usr/bin/env python3", "#!/usr/bin/python3", "#! /usr/bin/python3"]
+                  "#!/usr/bin/env python3", "#! /usr/bin/env python3", "#!/usr/bin/python3", "#! /usr/bin/python3"]
```

Comment:
Shebangs can have an effect on whether a code works or not (especially True if users mess up their environment). If I remember well, a while ago we agreed to only use the shebang `#!/usr/bin/env python3`. However, we also agreed to only update the shebang of files we were changing from that time onward. In other words, some old files might still have other shebangs, hence this test. If we want to avoid this being hardcoded I think this requires a separate task to go over all of our code base and unify this.

Reply:
@BenStarmerSmith @bdenoun what do you think then of having a set comparison of {'#!', '/usr/bin", 'python'} and adapting this line **any(fline in head for head in PYTHON_HEADERS)** to check if there's a match of at least 2 elements (assuming /usr/bin being part of /usr/bin/env). Might make it more robust to other combinations or using regex ('^#!/.*python' .) (https://unix.stackexchange.com/questions/66097/find-all-files-with-a-python-shebang)

Reply:
Using a regex Including python3 (let's not forget the 3) sounds like a very good idea

Reply:
I have added the Regex :)